scripts/bootstrap.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+from_source=""
+if [ "$1" = "-build" ]; then
+   from_source="t"
+fi
+
+set -e
+
+cd "$( dirname "$0" )/.."
+
+git fetch --tags --force --quiet
+version="$(git describe --tags "$(git rev-list --tags --max-count=1)")"
+os="$(uname)"
+arch="$(uname -m)"
+if [ "$from_source" = "" ]; then
+    if [ "$os" = "Darwin" ]; then
+	os="macos"
+    elif [ "$os" = "Linux" ]; then
+	os="linux"
+    else
+	echo "Unsupported OS."
+	exit 1
+    fi
+
+    curl -sLO "https://github.com/tigerbeetledb/tigerbeetle/releases/download/$version/tigerbeetle-$arch-$os-$version.zip"
+    unzip -qo "tigerbeetle-$arch-$os-$version.zip"
+else
+    git checkout "$version"
```

Comment:
One way that I think is safe to do this on Linux (although, it won't echo the final message...):
```
exec /bin/sh -c "git checkout \"$version\" && ./scripts/install.sh"
```

Reply:
(there are some others [here](https://stackoverflow.com/questions/2285403/how-to-make-shell-scripts-robust-to-source-being-changed-as-they-run/2358432#2358432))

Reply:
Would `set -e` also be needed here?

Reply:
Hmm... Showing the end message would be good though...

Reply:
Yeah - I think there are better methods in the SO post I linked above. We should be able to put it in a block and add the echo, then exit after the block to be safe (or wrap the whole script in a block)

Reply:
:+1: done thanks!